src/main/java/com/griddynamics/cd/nrp/internal/uploading/impl/ArtifactUpdateApiClientImpl.java

```diff
@@ -119,14 +151,14 @@ public void onComplete(Future<RestResponse> future) throws InterruptedException
 
     /**
      * Creates jersey HTTP client
-     * @param login Username on the remote server
+     *
+     * @param login    Username on the remote server
      * @param password User's password
      * @return HTTP client
      */
     private Client getClient(String login, String password) {
         ClientConfig config = new DefaultClientConfig();
         Client client = Client.create(config);
-        client.setExecutorService(asyncRequestsExecutorService);
```

Comment:
Yes, so I'll revert this change and there will be 2 executors:
* first only for this Client
* second for ArtifactMetaInfos from fileBockingQueue and executing this.sendRequest()